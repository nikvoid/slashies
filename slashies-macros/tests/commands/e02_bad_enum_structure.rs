use slashies_macros::Commands;

#[derive(Commands)]
enum BadCommands {
    HasNamedFields { named_field: u64 },
}

#[derive(Commands)]
enum BadCommands2 {
    HasNoFields,
}

#[derive(Commands)]
enum BadCommands3 {
    HasMultipleUnnamedFields(u64, u64),
}

fn main() {}
