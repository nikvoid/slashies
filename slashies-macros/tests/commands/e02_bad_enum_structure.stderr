error: Variants of a Commands enum must have exactly one unnamed field
 --> tests/commands/e02_bad_enum_structure.rs:5:5
  |
5 |     HasNamedFields { named_field: u64 },
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Variants of a Commands enum must have exactly one unnamed field
  --> tests/commands/e02_bad_enum_structure.rs:10:5
   |
10 |     HasNoFields,
   |     ^^^^^^^^^^^

error: Variants of a Commands enum must have exactly one unnamed field
  --> tests/commands/e02_bad_enum_structure.rs:15:5
   |
15 |     HasMultipleUnnamedFields(u64, u64),
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
